랜딩 페이지 구현

SH9480P opened the pull request:
## 작업 개요

closes #268

## 작업 상세 내용

- 랜딩 페이지 디자인
  - 배너
  - 기능 소개
  - 학습법
  - 퀴즈 페이지 가이드
  - 실시간 시각화 가이드
- 배너에 `지금 시작하기` 버튼 추가
  - 사이드바의 `학습 시작하기` 버튼을 클릭하고, `Docker란?` 페이지로 이동하는 효과.
  - `학습 시작하기` 버튼을 클릭하기 위해, 해당 DOM Element를 참조하는 ref를 App 컴포넌트로 끌어올리고, `학습 시작하기`와 `지금 시작하기`에 props drilling해야 했음.

## 문제점 혹은 고민

~~(아직 구현이 덜 된 감이 있지만.. 일단 피드백 받기 위해 PR 올려봅니다)~~

1. 상단 배너에 `지금 시작하기` 버튼을 넣으면 어떨까요? 버튼을 누르면 학습 시작하기가 눌리고, Docker란? 페이지로 이동하게 만들면 어떨까 싶습니다.
2. Landing Page에 시각 자료가 필요할까요?
3. Landing Page에 어떤 내용을 더 넣으면 좋을까요?

SH9480P committed:
feat: 랜딩 페이지 최초 디자인

tuchongkim commented on frontend/src/components/staticpages/LandingPage.tsx:
```diff
@@ -1,23 +1,144 @@
+                    <div className='flex-1 bg-white rounded-xl shadow-md overflow-hidden'>
+                        <div className='bg-blue-50 p-6'>
+                            <div className='flex items-center gap-3 mb-4'>
+                                <Terminal className='w-8 h-8 text-blue-600' />
```
👍

guune commented:
랜딩 페이지 깔끔하게 만들어 주신 것 같습니다. 반응형까지 고려 하셨군요!

> 상단 배너에 지금 시작하기 버튼을 넣으면 어떨까요? 버튼을 누르면 학습 시작하기가 눌리고, Docker란? 페이지로 이동하게 만들면 어떨까 싶습니다.

괜찮은 것 같습니다.

> Landing Page에 시각 자료가 필요할까요?
> Landing Page에 어떤 내용을 더 넣으면 좋을까요?

지금 생각나는 것은 시각화에 대한 설명(컨테이너 상태를 어떻게 표현하고, docker run 했을때 어떻게 화살표로 표현 되는지)이 생각나네요
그런데 막상 들어가면 지금 깔끔한 페이지가 복잡해질 것 같습니다. 어렵네요

tuchongkim commented:
> 2. Landing Page에 시각 자료가 필요할까요?
> 3. Landing Page에 어떤 내용을 더 넣으면 좋을까요?

만약 추가한다면 퀴즈 페이지 화면을 맨 아래에 넣어서 각 요소의 위치?에 대해서 설명하면 될 것 같긴 합니다

SH9480P committed:
refactor: landing page banner border rounded

SH9480P committed:
refactor: 반복되는 tag를 Component로 분리

SH9480P committed:
feat: 퀴즈 페이지 가이드, 시각화 가이드 추가

SH9480P committed:
feat: Landing Page에 지금 시작하기 버튼 기능 구현

- 지금 시작하기 버튼 클릭 시, side bar의 학습 시작하기 버튼을 클릭
  - 학습 시작하기 DOM Element를 가져오기 위해, 학습 시작하기 버튼에 ref를 등록
  - 등록된 ref를 학습 시작하기와 지금 시작하기 두 곳에서 사용해야 하므로, startButtonRef를 공통 부모 요소인 App.tsx로 끌어올렸음
- Docker란? 페이지로 navigate

tuchongkim commented on frontend/src/components/staticpages/LandingPage.tsx:
```diff
@@ -1,23 +1,292 @@
-                </p>
-            </section>
+            <Banner startButtonRef={startButtonRef} />
+
+            <FeatureIntro />
+            <hr />
+
+            <StudyGuide />
+            <hr />
+
+            <QuizPageGuide />
+            <hr />
```
혹시 이 부분에서 hr 을 화면 전체로 보이게 해주신 것 같은데 의도하신 걸까요?

SH9480P replied:
화면 전체로 보인다는게 무슨 뜻인가요?
사이드바와 헤더를 제외한 영역에 구분선이 그어지도록 만들었습니다.

tuchongkim commented on frontend/src/components/staticpages/LandingPage.tsx:
```diff
@@ -1,23 +1,292 @@
+                    </p>
+                    <button
+                        className='bg-Moby-Blue text-white px-8 py-3 rounded-lg font-semibold hover:bg-blue-800 transition-colors'
+                        onClick={() => {
+                            startButtonRef.current?.click();
+                            navigate('/what-is-docker');
+                        }}
```
좋습니다!

guune commented on frontend/src/assets/visualization-demo.gif:
예전 gif 라서 나중에 수정이 필요해 보입니다. 제가 readme에 시각화 gif 추가하면서 수정하도록 할께요

guune commented on frontend/src/components/staticpages/LandingPage.tsx:
```diff
@@ -1,23 +1,292 @@
+);
+
+const QuizPageGuide = () => {
+    const imageAlt = '번호가 매겨진 퀴즈 페이지의 요소';
+    return (
+        <div className='container mx-auto px-4 py-16'>
+            <h2 className='text-3xl mb-6'>퀴즈 페이지 가이드</h2>
```
가이드를 제 생각보다 더 좋게 해주셨네요 👍

guune commented on frontend/src/components/staticpages/LandingPage.tsx:
```diff
@@ -32,23 +37,37 @@ const LandingPage = () => {
+                    <button
+                        className='bg-Moby-Blue text-white px-8 py-3 rounded-lg font-semibold hover:bg-blue-800 transition-colors'
+                        onClick={() => {
+                            startButtonRef.current?.click();
```
이런식으로 되는거 신기하네요 ref로 dom 조작하는 예시 잘 봤습니다
